x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -189,6 +195,23 @@ export function InnerWorkspacePanel({
     }
   }, [expression, localState.expressionBuildError]);
 
+  if (visualizeTriggerFieldContext && !localState.visualizeTriggerSuggestionsLoaded && !title) {
+    const suggestions = getSuggestions({
+      datasourceMap,
+      datasourceStates,
+      visualizationMap,
+      activeVisualizationId,
+      visualizationState,
+      visualizeTriggerFieldContext,
+    });
+    if (suggestions.length) {
+      const selectedSuggestion =
+        suggestions.find((s) => s.visualizationId === activeVisualizationId) || suggestions[0];
+      switchToSuggestion(dispatch, selectedSuggestion, 'SWITCH_VISUALIZATION');
+      setLocalState((s) => ({ ...s, visualizeTriggerSuggestionsLoaded: true }));
+    }
+  }
+
   function onDrop() {
     if (suggestionForDraggedField) {
       trackUiEvent('drop_onto_workspace');
```

Comment:
While it's loading, the call to action rendered by `renderEmptyWorkspace` is flashing briefly - if the server takes a while to respond, this can take quite long. That state is confusing for the user ("why do I have to drag something, I clicked visualize on that field"). I would suggest just not rendering anything while waiting for an initial context.

Reply:
You are def right! I should have thought it, thanx will fix it